app/lib/server/utilities/filterParser.js

```diff
@@ -0,0 +1,173 @@
+const arrayRelationalConditions = new Set(['between', 'notBetween', 'in', 'notIn']);
+const relationalOperators = new Set([...arrayRelationalConditions, ...unitaryRelationalOperators]);
+const logicOperators = new Set(['not', 'and', 'or']);
+
+const _groupOperatorName = '_goperator';
+const defaultGroupOperator = 'and';
+
+const reservedNames = new Set([...relationalOperators, ...logicOperators, ...[_groupOperatorName]]);
+
+const transformationSentinel = 'and';
+
+class TransformHelper {
+    constructor(opts) {
+        this.opts = opts;
+    }
+
+    insertSentinel(filter) {
+        return { [transformationSentinel]: filter };
+    }
+
+    removeSentinel(filter) {
+        return filter[Op[transformationSentinel]];
+    }
+
+    handleIntermidiateFilterNode(group, groupOperator) {
+        const transformedSubfilter = this.transformHelper(group);
+        return this.opts.pruneRedundantANDOperator && groupOperator === defaultGroupOperator ?
+            transformedSubfilter :
+            { [Op[groupOperator]]: transformedSubfilter };
+    }
+
+    handelFieldFilterTail(fieldGroup, groupOperator) {
+        const transformedFieldGroup = Object.fromEntries(Object.entries(fieldGroup)
+            .map(([relOp, val]) => [
+                Op[relOp] ?? throwWrapper(new Error(`No relational operator like <${relOp}>, only <${[...relationalOperators]}>`)),
+                val,
+            ]));
+        return this.opts.pruneRedundantANDOperator && groupOperator === defaultGroupOperator ?
+            transformedFieldGroup :
+            { [Op[groupOperator]]: transformedFieldGroup };
+    }
+
+    pullGroupOperator(group) {
+        const readGroupOperator = group[_groupOperatorName];
+        const groupOperator = readGroupOperator ?? defaultGroupOperator;
+        delete group[_groupOperatorName];
+        return groupOperator;
+    }
+
+    transformHelper(filter) {
+        return Object.fromEntries(Object.entries(filter)
+            .map(([k, group]) => {
+                const groupOperator = this.pullGroupOperator(group);
+                if (!reservedNames.has(k)) { // Assumes that k is field from db view
+                    return [k, this.handelFieldFilterTail(group, groupOperator)];
+                } else { // Then k stands for logical operator
+                    return [Op[k], this.handleIntermidiateFilterNode(group, groupOperator)];
+                }
+            }));
+    }
+
+    transform(filter) {
+        if (!filter) {
+            return {};
+        }
+        filter = this.insertSentinel(filter);
+        filter = this.transformHelper(filter);
+        filter = this.removeSentinel(filter);
+        return filter;
+    }
+}
+
+/**
+ * Transform filter object from http request.query to sequelize where-clause object
+ * Considering that the filter object as a tree, then each root -> leaf path can be described as ,
```

Comment:
`Considering the filter object is a tree, each root -> leaf path can be described as`

Reply:
ok